### PR TITLE
Add Skip Releases Option

### PR DESCRIPTION
**Pull request by oakeyc**
Closes https://github.com/github/octoshift/issues/4226.

- [x] Did you write/update appropriate tests
- [ ] ~~Release notes updated (if appropriate)~~ (see https://github.com/github/gh-gei/pull/329#issuecomment-1090954211)
- [x] Appropriate logging output
- [x] Issue linked: https://github.com/github/octoshift/issues/4226

### CODE REVIEW

**Comment by synthead** [comment]
Hey @dylan-smith, do you think we should add some release notes for this contribution?  This is being shipped as a hidden feature for now, so maybe the release notes should be added when it becomes un-hidden?  What do you think?

**Comment by dylan-smith** [comment]
I agree with that. Release notes when we unhide it.

**Review by dylan-smith — CHANGES_REQUESTED** [review]

File: src/OctoshiftCLI.Tests/gei/Commands/GenerateScriptCommandTests.cs [hunk 1]
```diff
@@ -620,6 +621,22 @@ public async Task It_Uses_Ado_Pat_When_Provided()
             mockEnvironmentVariableProvider.VerifyNoOtherCalls();
         }
 
+        [Fact]
+        public void It_Uses_Skip_Releases_When_Provided()
```
Comment (ArinGhazarian) [inline src/OctoshiftCLI.Tests/gei/Commands/GenerateScriptCommandTests.cs:625]: just a nit rename recommendation:
```suggestion
        public void It_Adds_Skip_Releases_To_Migrate_Repo_Command_When_Provided()
```

**Review by ArinGhazarian — COMMENTED** [review]
Just a small nit, the rest is 💯

File: src/gei/Commands/GenerateScriptCommand.cs [hunk 2]
```diff
@@ -162,6 +171,8 @@ public GenerateScriptCommand(OctoLogger log, ISourceGithubApiFactory sourceGithu
                 }
             }
 
+            _log.LogInformation($"SKIP RELEASES: {skipReleases}");
```
[the remark this answers is not in the document]
Comment (synthead) [reply src/gei/Commands/GenerateScriptCommand.cs]: Updated with 65c9999473c76390f9325f7f1c4398a964863044!

File: src/gei/Commands/GenerateScriptCommand.cs [hunk 3]
```diff
@@ -61,6 +61,12 @@ public GenerateScriptCommand(OctoLogger log, ISourceGithubApiFactory sourceGithu
                 IsRequired = false,
                 Description = "Only effective if migrating from GHES. Disables SSL verification when communicating with your GHES instance. All other migration steps will continue to verify SSL. If your GHES instance has a self-signed SSL certificate then setting this flag will allow data to be extracted."
             };
+            var skipReleases = new Option("--skip-releases")
+            {
+                IsHidden = true,
+                IsRequired = false,
+                Description = "Extra setting to skip releases when migrating. Default value is false."
```
[the remark this answers is not in the document]
Comment (synthead) [reply src/gei/Commands/GenerateScriptCommand.cs]: Fixed with 1d633a12c5059617481f64fb3ff26e560266f2c0!

File: src/gei/Commands/MigrateRepoCommand.cs [hunk 4]
```diff
@@ -92,7 +92,12 @@ public MigrateRepoCommand(OctoLogger log, ISourceGithubApiFactory sourceGithubAp
                 IsRequired = false,
                 Description = "An authenticated SAS URL to an Azure Blob Storage container with a pre-generated metadata archive. Only used when an archive has been generated and uploaded prior to running a migration (not common). Must be passed in when also using --git-archive-url"
             };
-
+            var skipReleases = new Option("--skip-releases")
+            {
+                IsHidden = true,
+                IsRequired = false,
+                Description = "Extra setting to skip releases when migrating. Default value is false."
```
[the remark this answers is not in the document]
Comment (synthead) [reply src/gei/Commands/MigrateRepoCommand.cs]: Fixed with 1d633a12c5059617481f64fb3ff26e560266f2c0!

File: src/OctoshiftCLI.Tests/gei/Commands/GenerateScriptCommandTests.cs (hunk 1 above)
Comment (synthead) [reply src/OctoshiftCLI.Tests/gei/Commands/GenerateScriptCommandTests.cs]: Updated in 5410d763c4be7c97221ebf2828edcda6d2326a56!

File: src/OctoshiftCLI.Tests/gei/Commands/GenerateScriptCommandTests.cs [hunk 5]
```diff
@@ -620,6 +621,22 @@ public async Task It_Uses_Ado_Pat_When_Provided()
             mockEnvironmentVariableProvider.VerifyNoOtherCalls();
         }
 
+        [Fact]
+        public void It_Adds_Skip_Releases_To_Migrate_Repo_Command_When_Provided()
+        {
+            var repo = "foo-repo";
+            var repos = new List<string>() { repo };
+
+            var command = new GenerateScriptCommand(new Mock<OctoLogger>().Object, null, null, null);
```
Comment (dylan-smith) [inline src/OctoshiftCLI.Tests/gei/Commands/GenerateScriptCommandTests.cs:630]:
```suggestion
            var command = new GenerateScriptCommand(TestHelpers.CreateMock<OctoLogger>().Object, null, null, null);
```

File: src/OctoshiftCLI.Tests/gei/Commands/MigrateRepoCommandTests.cs [hunk 6]
```diff
@@ -849,7 +860,49 @@ public async Task It_Falls_Back_To_Github_Target_Pat_If_Github_Source_Pat_Is_Not
...
+                It.IsAny<bool>()));
+        }
+
+        [Fact]
+        public async Task It_Skips_Releases_When_Option_Is_True()
+        {
+            // Arrange
+            var environmentVariableProviderMock = new Mock<EnvironmentVariableProvider>(null);
+
+            var mockGithub = new Mock<GithubApi>(null, null, null);
```
Comment (dylan-smith) [inline src/OctoshiftCLI.Tests/gei/Commands/MigrateRepoCommandTests.cs:873]:
```suggestion
            var mockGithub = TestHelpers.CreateMock<GithubApi>();
```

File: src/OctoshiftCLI.Tests/gei/Commands/MigrateRepoCommandTests.cs [hunk 7]
```diff
@@ -849,7 +860,49 @@ public async Task It_Falls_Back_To_Github_Target_Pat_If_Github_Source_Pat_Is_Not
...
+            var mockGithubApiFactory = new Mock<ITargetGithubApiFactory>();
+            mockGithubApiFactory.Setup(m => m.Create(It.IsAny<string>(), It.IsAny<string>())).Returns(mockGithub.Object);
+
+            var actualLogOutput = new List<string>();
+            var mockLogger = new Mock<OctoLogger>();
```
Comment (dylan-smith) [inline src/OctoshiftCLI.Tests/gei/Commands/MigrateRepoCommandTests.cs:878]:
```suggestion
            var mockLogger = TestHelpers.CreateMock<OctoLogger>();
```

**Review by ArinGhazarian — APPROVED** [review]

File: src/OctoshiftCLI.Tests/gei/Commands/MigrateRepoCommandTests.cs (hunk 7 above)
Comment (synthead) [reply src/OctoshiftCLI.Tests/gei/Commands/MigrateRepoCommandTests.cs]: Updated in 3ddc2374b698be4f5b2e1a721c85fed6cddf2ec6!

File: src/OctoshiftCLI.Tests/gei/Commands/MigrateRepoCommandTests.cs (hunk 6 above)
Comment (synthead) [reply src/OctoshiftCLI.Tests/gei/Commands/MigrateRepoCommandTests.cs]: Updated in 3ddc2374b698be4f5b2e1a721c85fed6cddf2ec6!

File: src/OctoshiftCLI.Tests/gei/Commands/GenerateScriptCommandTests.cs (hunk 5 above)
Comment (synthead) [reply src/OctoshiftCLI.Tests/gei/Commands/GenerateScriptCommandTests.cs]: Updated in 3ddc2374b698be4f5b2e1a721c85fed6cddf2ec6!

**Review by dylan-smith — APPROVED** [review]

**Pull request merged**